eo-maven-plugin/src/main/java/org/eolang/maven/fp/Footprint.java

```diff
@@ -0,0 +1,198 @@
+ *  4) copy content from cache to target and return target</p>
+ * @since 0.41.0
+ */
+public final class Footprint implements Func<Func<Path, String>, Path> {
+    /**
+     * Absolute path to file with source code.
+     */
+    private final Path source;
+
+    /**
+     * Absolute path to target file.
+     */
+    private final Path target;
+
+    /**
+     * Global cache.
+     */
+    private final Cache cache;
+
+    /**
+     * Ctor.
+     * @param source Absolute source path
+     * @param target Absolute target path
+     * @param cache Cache
+     */
+    public Footprint(
+        final Path source,
+        final Path target,
+        final Cache cache
+    ) {
+        this.source = source;
+        this.target = target;
+        this.cache = cache;
+    }
+
+    @Override
+    public Path apply(final Func<Path, String> content) throws Exception {
+        if (!Footprint.exists(this.source)) {
+            throw new IllegalStateException(
+                String.format("Source file %s does not exist", this.source)
+            );
+        }
+        final Path result;
+        if (Footprint.exists(this.target) && Footprint.isAfter(this.target, this.source)) {
+            result = this.target;
+        } else {
+            result = this.dependentOnCache(content);
+        }
+        return result;
+    }
+
+    /**
+     * Updates target depends on global cache.
+     * @param content Content function
+     * @return Path to target
+     * @throws IOException If fails to check files last modified time
+     */
+    private Path dependentOnCache(final Func<Path, String> content) throws Exception {
+        final Path result;
+        if (this.cache.cacheable()) {
+            final Path cached = this.cache.path();
+            if (Footprint.exists(cached) && Footprint.isAfter(cached, this.source)) {
+                result = this.copiedFromCache();
+            } else {
+                result = this.updatedAll(content);
+            }
+        } else {
+            result = this.updatedTarget(content);
+        }
+        return result;
+    }
+
+    /**
+     * Takes content given function and update target file.
+     * @param content Content function
+     * @return Absolute path to target file
+     */
+    private Path updatedTarget(final Func<Path, String> content) throws Exception {
+        return Footprint.updated(this.source, this.target, content);
+    }
+
+    /**
+     * Takes content from given function, updates target file, updates global cache file.
+     * @param content Content function
+     * @return Absolute path to target file
+     * @throws Exception If failed to extract content or update files
+     */
+    private Path updatedAll(final Func<Path, String> content) throws Exception {
+        final Func<Path, String> sticky = new StickyFunc<>(content);
+        final Path result = this.updatedTarget(sticky);
+        Footprint.updated(this.source, this.cache.path(), sticky);
+        return result;
```

Comment:
@maxonfjvipon same here: you use OOP and then procedural style. Such a mix is confusing and makes code longer. Make `updatedAll` return `Provider<Path>`. Also, make `copiedFromCache` already return `Provider<Path>`. Then, you will be able to compose them in one larger function/object. The size of the code will be smaller and the names of the methods will better represent what they do: their results will really act like "path that is updated in both places".